src/main/java/calculator/StringAddCalculator.java

```diff
@@ -0,0 +1,32 @@
+package calculator;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Stream;
+
+public class StringAddCalculator {
+    public static int splitAndSum(String text) {
+        if(isNullOrEmpty(text)){ return 0; }
+        return sum(split(text));
+    }
+    private static String[] split(String text){
+        Matcher m = Pattern.compile("//(.)\n(.*)").matcher(text);
+        if (m.find()) {
+            return m.group(2).split(m.group(1));
+        }
+        return text.split("[,:]");
+    }
+    private static int sum(String[] numbers){
+        return Stream.of(numbers)
+                .mapToInt(Integer::parseInt)
+                .peek(StringAddCalculator::isValid)
+                .sum();
+    }
+    private static void isValid(int number){
+        if(number < 0)
+            throw new RuntimeException();
```

Comment:
```suggestion
            throw new IllegalArgumentException();
```

RuntimeException은 의미가 너무 포괄적이라고 생각이 듭니다.
IllegalArgumentException과 같이 더 구체적인 예외를 던지는게 더 나을 것 같아요!